test/function.localstack.test.ts

```diff
@@ -51,7 +54,11 @@ interface TestFunctionBase {
     expected: OO extends void
       ? null
       : OO | ((context: Outputs) => OO extends void ? null : O),
-    payload?: I | ((context: Outputs) => I)
+    payload?: I | ((context: Outputs) => I),
+    /**
+     * When true uses jest's {@link jest.Matchers.toMatchObject} logic instead of {@link jest.Matchers.toEqual}.
+     */
+    matchObject?: boolean
```

Comment:
Could pass the reference to the function instead.

Reply:
Ended up just repurposing the expected method to accept the payload and run arbitrary logic.

```ts
expected: T | (context, payload) => void
```

Where a value will just use `.toEquals`